packages/components/inputs/select-utils/src/types.ts

```diff
@@ -0,0 +1,15 @@
+import type SelectComponents from 'react-select';
+
+type ComponentProps<T> = T extends React.ComponentType<infer P> ? P : never;
```

Comment:
This is a conditional type that extracts the props type from a React component type.